language/libra-tools/libra-events-fetcher/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use libra_json_rpc_client::{views::EventView, JsonRpcBatch, JsonRpcClient, JsonRpcResponse};
+use libra_types::{
+    account_address::AccountAddress,
+    account_state::AccountState,
+    account_state_blob::AccountStateBlob,
+    event::{EventHandle, EventKey},
+};
+use reqwest::Url;
+use std::convert::TryFrom;
+
+const BATCH_SIZE: u64 = 500;
+
+pub struct LibraEventsFetcher(JsonRpcClient);
+
+impl LibraEventsFetcher {
+    pub fn new(url: &str) -> Result<Self> {
+        let url = Url::parse(url)?;
+        Ok(Self(JsonRpcClient::new(url)?))
+    }
+
+    fn execute_single_command(&self, cmd: JsonRpcBatch) -> Result<JsonRpcResponse> {
+        let mut ret = self.0.execute(cmd)?;
+        if ret.len() != 1 {
+            bail!("Unexpected response length")
+        }
+        ret.pop().unwrap()
+    }
+
+    fn get_account_state(&self, account: AccountAddress) -> Result<Option<AccountState>> {
+        let account_state: AccountStateBlob = {
+            let mut batch = JsonRpcBatch::new();
+            batch.add_get_account_state_with_proof_request(account, None, None);
```

Comment:
Call get_account instead of this get_account_state_with_proof

Reply:
It will have to be `get_account ` unfortunately because we need the event count, which is not included in `jsonrpc::Account`..